spectator/counter.py

```diff
@@ -13,9 +13,13 @@ def __init__(self, meter_id: MeterId,
         self._writer = writer
 
     def increment(self, delta: int = 1) -> None:
+        if self.meterId.error() != MeterIdError.NONE:
+            self._writer.write(self.errorString, 1)
```

Comment:
This will increment the dropped measurements counter for every update of the meter rather than once for each publish attempt. For example, say a single counter has issues and is updated roughly 1k times per second. SpectatorD will update the dropped measurements counter on publish attempts, so say every 5s it would update it by 1 and it would show ~12 dropped measurements per minute. With this approach the dropped measurements counter will show 1k measurements/second dropped which may lead to an exaggerated view of the impact.

It is probably fine and I don't have a better suggestion, but it might be another reason why it would be better to have most validation checks done later in spectatord. One alternative might be to do it once when the meter is created, but I think that could be more confusing. If the user caches the meter and reuses, then it would only increment once and may not get noticed if it runs for a long time. If the meter is created for each update, then it would have the same behavior as now. Having inconsistent behavior depending on usage could increase user confusion.

Another option might be to do something like use a special value for the not a string case, e.g. `[not a string]`, that would be accepted by spectatord, but fail validation so spectatord would update the counter based on the server response and the view of the impact would be consistent.

Reply:
Hm, there's a bit to think about here and no clear winners, although I am onboard with dropping the extra validation checks and letting SpectatorD handle most of that work.

SpectatorD does invalid character rewriting, so using `[not a string]` as a tag value would not work:

```
echo "c:testInvalid,id=[not a string]:1" | nc -u -w0 localhost 1234
```

```
name,testInvalid,:eq,id,_not_a_string_,:eq,:and
```

We could maybe use length to trigger a validation error:

```
echo "c:testInvalid,id=:1" | nc -u -w0 localhost 1234
```

```
[2022-08-25 18:09:46.057] [spectator] [info] Validation error: value too short: id = [] (0 < 1) (tags={"name":"testInvalid","id":"","nf.account":"...","nf.app":"...","nf.asg":"...","nf.cluster":"...","nf.node":"...","nf.region":"...","nf.stack":"...","nf.vmtype":"...","nf.zone":"..."})
```

This sort of gets back to the original idea of filtering out not-a-str tags, but it prevents the metric from existing, produces a SpectatorD error message and triggers a `spectator.measurements` increment.